File: pytools/examples/java/client/protocol/MsgRoleBase.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class MsgRoleBase
{
	private int uid;
	private String uname;


	public MsgRoleBase()
	{

	}

	public MsgRoleBase(Packet packet)
	{
		uid = packet.readInt();
		uname = packet.readString();
	}

	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeInt(uid);
		packet.writeString(uname);
		return packet.readBytes();
	}

	public byte[] getBytes()
	{
		return this.encode();
	}



	public void setUid(int uid)
	{
		this.uid = uid;
	}
	public long getUid()
	{
		return PacketUtil.readUInt(this.uid);
	}

	public void setUname(String uname)
	{
		this.uname = uname;
	}
	public String getUname()
	{
		return this.uname;
	}

}

File: pytools/examples/csharp/client/Protocol/MsgRoleBase.cs
using System.Collections;
using System.Collections.Generic;


public class MsgRoleBase
{
	private uint _uid;
	private string _uname;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteUint(this._uid);
		packet.WriteString(this._uname);
		return packet;
	}

	public MsgRoleBase()
	{
	}

	public MsgRoleBase(Packet packet)
	{
		this._uid = packet.ReadUint();
		this._uname = packet.ReadString();
	}

	public List<byte> GetBuffer()
	{
		return this.Encode().GetBuffer();
	}


	public uint uid
	{
		get { return this._uid; }
		set { this._uid = value; }
	}

	public string uname
	{
		get { return this._uname; }
		set { this._uname = value; }
	}

}
